python/pyspark/cloudpickle.py

```diff
@@ -237,28 +262,14 @@ def dump(self, obj):
             if 'recursion' in e.args[0]:
                 msg = """Could not pickle object as excessively deep recursion required."""
                 raise pickle.PicklingError(msg)
-        except pickle.PickleError:
-            raise
-        except Exception as e:
-            emsg = _exception_message(e)
-            if "'i' format requires" in emsg:
-                msg = "Object too large to serialize: %s" % emsg
-            else:
-                msg = "Could not serialize object: %s: %s" % (e.__class__.__name__, emsg)
-            print_exec(sys.stderr)
-            raise pickle.PicklingError(msg)
-
```

Comment:
This exception handling is Spark specific, it has been moved to serializers.py `CloudPickleSerializer.dumps`

Reply:
I'm glad this is moved, should make the next update easier.